src/controllers/api-key.js

```diff
@@ -67,25 +199,50 @@ function ApiKeyController(context) {
     const { pathInfo: { headers }, params: { id } } = requestContext;
     const imsOrgId = headers['x-ims-gw-org-id'];
 
-    validateImsOrgId(imsOrgId);
-    if (id) {
-      const error = new ErrorWithStatusCode('Delete API key not implemented', 501);
+    try {
+      const imsUserToken = getImsUserToken(headers);
+      await validateImsOrgId(imsOrgId, imsUserToken);
+      const apiKeyEntity = await dataAccess.getApiKeyById(id);
+      const { authInfo: { profile } } = attributes;
+
+      // Currently the email is assigned as the imsUserId
+      const imsUserId = profile.email;
+      if (apiKeyEntity.getImsUserId() !== imsUserId || apiKeyEntity.getImsOrgId() !== imsOrgId) {
+        throw new ErrorWithStatusCode('Invalid request: API key not found', STATUS_FORBIDDEN);
```

Comment:
Since the message is not found, should we return a 404?

Reply:
I think we do not want to allow a user who did not create this api key to delete it. This `if` statement was to check that. I could change the message here.
I will add another check for the API Key not found scenario.